apps/api/src/app/db/ratings/ratings.store.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { RatingsDocument } from './ratings.schema';
+import { JwtPayload } from '@dragonfish/shared/models/auth';
+import { isNullOrUndefined } from '@dragonfish/shared/functions';
+import { RatingOption } from '@dragonfish/shared/models/reading-history';
+import { ContentStore } from '../content';
+
+@Injectable()
+export class RatingsStore {
+    constructor(
+        @InjectModel('Ratings') private readonly ratings: Model<RatingsDocument>,
+        private readonly contentStore: ContentStore,
+    ) {}
+
+    public async addOrFetchRatingsDoc(user: JwtPayload, contentId: string) {
+        const existingDoc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+
+        if (isNullOrUndefined(existingDoc)) {
+            const newDoc = new this.ratings({
+                contentId: contentId,
+                userId: user.sub,
+            });
+
+            return newDoc.save();
+        } else {
+            return existingDoc;
+        }
+    }
+
+    public async addLike(user: JwtPayload, contentId: string) {
+        const doc = await this.ratings.findOne({ contentId: contentId, userId: user.sub });
+        doc.rating = RatingOption.Liked;
+
+        return doc.save().then(async (doc) => {
+            await this.updateCounts(contentId);
+            return doc;
+        });
```

Comment:
Avoid callbacks, especially when we have `async` available.
```suggestion
        const savedDoc = await doc.save();
        await this.updateCounts(contentId);
        return savedDoc;
```